### ISSUE TITLE
_TZ3000_dfgbtub0

### ISSUE THREAD

**Issue by ReichertD** [dresden-elektronik/deconz-rest-plugin#4890]
## Device

- Product name: Scene Switch
- Manufacturer: _TZ3000_dfgbtub0
- Model identifier: TS0042
- Device type : 
  - Switch

[Moes Scene Switch at Amazon](https://www.amazon.de/gp/product/B08XJTJ379/ref=ppx_yo_dt_b_asin_title_o00_s00?ie=UTF8&psc=1)

## Screenshots

Cluster info:
First time add new switch (same switch as second and third time)
 
![image](https://user-images.githubusercontent.com/24540342/118637598-485ef880-b7d6-11eb-9fef-ea60ab1f0c38.png)

Second time add new switch (same switch as first and third time) and klicking a button
 
![image](https://user-images.githubusercontent.com/24540342/118637612-4dbc4300-b7d6-11eb-85fd-aee6ebf2568c.png)

Third time add new switch (same switch as in first and second time)
 
![image](https://user-images.githubusercontent.com/24540342/118637624-51e86080-b7d6-11eb-9ee8-9a756164c82d.png)

Node info:
 
![image](https://user-images.githubusercontent.com/24540342/118637630-544aba80-b7d6-11eb-91bf-3946bbfcfeac.png)

API Info Switch:
{
    "config": {
        "battery": null,
        "on": true,
        "reachable": true
    },
    "ep": 1,
    "etag": "516d6483ed4eb2cc7445852637b76551",
    "lastseen": "2021-05-18T10:25Z",
    "manufacturername": "_TZ3000_dfgbtub0",
    "mode": 1,
    "modelid": "TS0042",
    "name": "Schalter",
    "state": {
        "buttonevent": null,
        "lastupdated": "none"
    },
    "type": "ZHASwitch",
    "uniqueid": "84:71:27:ff:fe:e2:f3:81-01-0006"
}

API Info Light 35 (automatically added during add switch)
{
    "etag": "fae6ebde3ad6a8e99334839d85c9e621",
    "hascolor": false,
    "lastannounced": null,
    "lastseen": "2021-05-18T10:25Z",
    "manufacturername": "_TZ3000_dfgbtub0",
    "modelid": "TS0042",
    "name": "On/Off light 35",
    "state": {
        "alert": "none",
        "on": false,
        "reachable": true
    },
    "swversion": null,
    "type": "On/Off light",
    "uniqueid": "84:71:27:ff:fe:e2:f3:81-01"
}

API Info Light 36 (automatically added during add switch)
{
    "etag": "3151ede34a3fea8d651dd66d3e320808",
    "hascolor": false,
    "lastannounced": null,
    "lastseen": "2021-05-18T10:25Z",
    "manufacturername": "_TZ3000_dfgbtub0",
    "modelid": "TS0042",
    "name": "On/Off light 36",
    "state": {
        "alert": "none",
        "on": false,
        "reachable": true
    },
    "swversion": null,
    "type": "On/Off light",
    "uniqueid": "84:71:27:ff:fe:e2:f3:81-02"
}

PhosconApp:
Switches:
![image](https://user-images.githubusercontent.com/24540342/118637937-9d027380-b7d6-11eb-84d3-62af1c2ee54a.png)

Lights:
![image](https://user-images.githubusercontent.com/24540342/118637995-a986cc00-b7d6-11eb-9307-3904e9093223.png)

**Comment by Smanar** [comment]
Hello, the result is "normal".

Do you have a linux machine to test a new code ?

**Comment by ReichertD** [comment]
Hi,

yes I heard so from the support team. ;)
I ran deconz and phoscon in a docker container on a raspberry. Will this help?

**Comment by Smanar** [comment]
>I ran deconz and phoscon in a docker container on a raspberry. Will this help?

lol, not for me, it s the worse situation ^^.
Not easy to compile the code on docker, you don't have a machine with full OS ? even a small raspberry.

**Comment by gfxdude-dus** [comment]
hi,
is this one the same? [amazon link](https://www.amazon.de/gp/product/B08LB9WKCJ/ref=ppx_yo_dt_b_asin_title_o01_s00?ie=UTF8&th=1) have issues to pair or get any answers. running on ubtuntu 20.

**Comment by Smanar** [comment]
Impossible to reconize a tuya device with a picture, the only way is including it on a zigbee network and check for the manufacture name.

**Comment by ReichertD** [comment]
I have a Windows running, this will be the easiest for me. Maybe I will be able to setup an older Raspberry. Will try this out and will come back to you to get the code/beta.

**Comment by Smanar** [comment]
You have the procedure here https://github.com/dresden-elektronik/deconz-rest-plugin#install-deconz-development-package-optional-linux-only
So for you, after having installed deconz :

```
sudo apt install deconz-dev
git clone --branch tuya_switch_4 https://github.com/Smanar/deconz-rest-plugin.git
cd deconz-rest-plugin
qmake && make -j2
sudo cp ../libde_rest_plugin.so /usr/share/deCONZ/plugins
```

If you have a small raspberry,it can freeze during compilation, my pi 3 do, so remove "-j2"

Remove the old entries before.

You will need to edit too the json file. It will be in /usr/share/deCONZ/devices/button_maps.json and the change will be https://github.com/Smanar/deconz-rest-plugin/commit/1fd60e9ec2cd3477a0dc2ec10c00ce351c81bb50#diff-a3adce467fe93b9c612c198c187dad34f5284d6c9adf66aa28127f0f619979b5

**Comment by ReichertD** [comment]
Hi, now I were able to do test some things.
I installed Ubunto VM and cloned your branch. After the make (need additionally "sudo apt install build-essentia") I updated the json, opened deCONZ and phoscon app and started a search for the switch. The switch were found and connected fast!
First I saw 2 on/off lights in phoscon, too. After a reboot these lights are gone in phoscon!

After that the switch is not usable in groups.
But(!) I connected this test phoscon bridge to my openhab and added the Switch to the things and created the channels. Now with your update I were able to get the events and states from the switch.

Looks like a good job!
